File: backend/src/main/java/com/regional/calendar/entity/Activity.java
package com.regional.calendar.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("activity")
public class Activity {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long festivalId;

    private Long regionId;

    private String regionName;

    private String name;

    private String nameLocal;

    private Integer type;

    private String subType;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    private Integer durationDays;

    private String location;

    private String address;

    private BigDecimal latitude;

    private BigDecimal longitude;

    private String transportation;

    private String parkingInfo;

    private String description;

    private String content;

    private String highlights;

    private String schedule;

    private String organizer;

    private String coOrganizer;

    private String sponsor;

    private String contactName;

    private String contactPhone;

    private String contactEmail;

    private String website;

    private String ticketInfo;

    private BigDecimal ticketPrice;

    private String ticketUrl;

    private Integer capacity;

    private Integer registeredCount;

    private Integer minAge;

    private Integer maxAge;

    private String weatherNote;

    private String dressCode;

    private String tips;

    private String imageUrl;

    private String bannerUrl;

    private String videoUrl;

    private String gallery;

    private String tags;

    private Long viewCount;

    private Long favoriteCount;

    private Long shareCount;

    private BigDecimal rating;

    private Integer ratingCount;

    private Integer sortOrder;

    private Integer isHot;

    private Integer isRecommended;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/lib/data/models/activity.dart
import 'package:json_annotation/json_annotation.dart';

part 'activity.g.dart';

@JsonSerializable()
class Activity {
	final int id;
	final int? festivalId;
	final int regionId;
	final String name;
	final String? nameLocal;
	final int type;
	final String? startTime;
	final String? endTime;
	final String? location;
	final String? address;
	final double? latitude;
	final double? longitude;
	final String? description;
	final String? content;
	final String? organizer;
	final String? contact;
	final String? ticketInfo;
	final String? imageUrl;
	final int status;

	Activity({
		required this.id,
		this.festivalId,
		required this.regionId,
		required this.name,
		this.nameLocal,
		this.type = 1,
		this.startTime,
		this.endTime,
		this.location,
		this.address,
		this.latitude,
		this.longitude,
		this.description,
		this.content,
		this.organizer,
		this.contact,
		this.ticketInfo,
		this.imageUrl,
		this.status = 1,
	});

	factory Activity.fromJson(Map<String, dynamic> json) => _$ActivityFromJson(json);
	Map<String, dynamic> toJson() => _$ActivityToJson(this);

	String get typeText {
		switch (type) {
			case 1: return '庆典';
			case 2: return '祭祀';
			case 3: return '比赛';
			case 4: return '表演';
			case 5: return '展览';
			default: return '其他';
		}
	}
}
